hack/test/unit

```diff
@@ -18,43 +18,51 @@ TESTDIRS="${TESTDIRS:-./...}"
 
 mkdir -p bundles
 
-cd api
-api_pkg_list=$(go list $TESTDIRS)
+case "$TESTDIRS" in
+"./api"* | "./...")
+	api_pkg_list=$(go -C ./api list ".${TESTDIRS#./api}")
+	;;
+esac
 if [ -n "${api_pkg_list}" ]; then
-	gotestsum --format=standard-quiet --jsonfile=../bundles/api-go-test-report.json --junitfile=../bundles/api-junit-report.xml -- \
+	gotestsum --format=standard-quiet --jsonfile=bundles/api-go-test-report.json --junitfile=bundles/api-junit-report.xml -- \
```

Comment:
This still needs to be run from the module root, I didn't see a `-C` equivalent for `gotestsum`

Reply:
Yeah, not 100% sure either; I think the go list returns the module path (e.g. "github.com/moby/moby/api/foo"), so maybe because of that it worked when I tried, but I should verify!

Reply:
The package list is derived from inside the module, the test should also be run from inside the module. if having a `go.work` makes go list return from multiple modules and is respected by the test runner, sure. To get this test working as you have it though, you need to cd into the module root.